mainPageCore/src/main/java/com/hamdan/forzenbook/mainpage/core/data/database/UserDao.kt

```diff
@@ -0,0 +1,33 @@
+package com.hamdan.forzenbook.mainpage.core.data.database
+
+import androidx.room.Dao
+import androidx.room.Insert
+import androidx.room.Query
+
+@Dao
+interface UserDao {
+
+    @Insert
+    suspend fun insert(userEntity: UserEntity)
+
+    @Query(
+        """
+            SELECT * FROM ${UserEntity.TABLE_NAME} WHERE ${UserEntity.USER_ID} = :userId
+        """
+    )
+    suspend fun getUser(userId: String): List<UserEntity>
+
+    @Query(
+        """
+            DELETE FROM ${UserEntity.TABLE_NAME} WHERE ${UserEntity.USER_ID} = :userId
+        """
+    )
+    suspend fun deleteUser(userId: String)
+
+    @Query(
+        """
+            DELETE FROM ${UserEntity.TABLE_NAME} WHERE ${UserEntity.ENTRY_DATE} <= (:currentTime - 86400000)
```

Comment:
What is this? lol

Reply:
this is to delete any old post information whenever a call to retrieve post info is made. It's time is set for 24 hours. it seemed like a cleaner way to just wipe any old posts from room than to keep makeing expensive checks on each post individualy as its asked for.

Reply:
Yeah but wtf is that ugly af hard coded number. Constant that shiz

Reply:
updated to use a constant.